api/src/main/java/com/datatorrent/api/Module.java

```diff
@@ -36,7 +36,7 @@
  * @since 3.3.0
  */
 @InterfaceStability.Evolving
-public interface Module
+public interface Module extends Operator
```

Comment:
The difference design could be having a common interface, which is shared by all type of Nodes in the DAG. Have updated the pull request with this approach.